src/phasorpy/io.py

```diff
@@ -522,6 +540,266 @@ def read_sdt(
+    channel: int | None = None,
+    keepdims: bool = True,
+) -> DataArray:
+    """Return image histogram and metadata from PicoQuant PTU T3 mode file.
+
+    PTU files contain time-correlated single photon counting measurement data
+    and instrumentation parameters.
+
+    Parameters
+    ----------
+    filename : str or Path
+        Name of PTU file to read.
+    selection : sequence of index types
+        Indices for all dimensions:
+
+        - ``None``: return all items along axis (default).
+        - ``Ellipsis``: return all items along multiple axes.
+        - ``int``: return single item along axis.
+        - ``slice``: return chunk of axis.
+          ``slice.step`` is binning factor.
+          If ``slice.step=-1``, integrate all items along axis.
+
+    trimdims : str
+        Axes to trim. The default is ``'TCH'``.
+    dtype : dtype-like
+        Unsigned integer type of image histogram array.
+        The default is ``uint16``. Increase the bit depth to avoid
+        overflows when integrating.
+    frame : int
+        If < 0, integrate time axis, else return specified frame.
+        Overrides ``selection`` for axis ``T``.
+    channel : int
+        If < 0, integrate channel axis, else return specified channel.
+        Overrides ``selection`` for axis ``C``.
+    keepdims :
+        If true (default), reduced axes are left as size-one dimension.
+
+    Returns
+    -------
+    xarray.DataArray
+        Decoded TTTR T3 records as up to 5-dimensional image array
+        with :ref:`axes codes <axes>` ``'TYXCH'`` and type specified
+        in ``dtype``:
+
+        - ``coords['H']``: times of the histogram bins.
+        - ``attrs['frequency']``: repetition frequency in MHz.
+
+    Raises
+    ------
+    ptufile.PqFileError
+        File is not a PicoQuant PTU file or is corrupted.
+    ValueError
+        File is not a PicoQuant PTU T3 mode file containing time-correlated
+        single photon counting data.
+
+    Examples
+    --------
+    >>> data = read_ptu(fetch('hazelnut_FLIM_single_image.ptu'))
+    >>> data.values
+    array(...)
+    >>> data.dtype
+    dtype('uint16')
+    >>> data.shape
+    (5, 256, 256, 1, 139)
+    >>> data.dims
+    ('T', 'Y', 'X', 'C', 'H')
+    >>> data.coords['H'].data
+    array(...)
+    >>> data.attrs['frequency']
+    2.51...
+
+    """
+    import ptufile
+
+    with ptufile.PtuFile(filename, trimdims=trimdims) as ptu:
+        if not ptu.is_t3 or not ptu.is_image:
+            raise ValueError(
+                f'{os.path.basename(filename)!r} '
+                'is not a PTU file containing a T3 mode image'
+            )
+        data = ptu.decode_image(
+            selection,
+            dtype=dtype,
+            frame=frame,
+            channel=channel,
+            keepdims=keepdims,
+            asxarray=True,
+        )
+
+    assert isinstance(data, DataArray)
+    data.attrs['frequency'] *= 1e-6  # MHz
+    return data
+
+
+def read_flif(
```

Comment:
Coudn't test this function because I haveno flif files

Reply:
There's a test file on Zenodo. This format is of no practical relevance. I just implemented it to get access to some old homodyne frequency-domain data.